src/game_initialization/create_engine.cpp

```diff
@@ -727,6 +727,15 @@ void create_engine::init_all_levels()
 	// Campaigns.
 	for(const config& data : game_config_.child_range("campaign"))
 	{
+		if(!data.has_attribute("id") || data["id"].empty()) {
+			if(!data.has_attribute("name") || data["name"].empty()) {
```

Comment:
technicailly `!data.has_attribute("name")` imples `data["name"].empty()` so this can be simplified

Reply:
Thanks - I missed that data was already `const`; will do the simplification.

The build error is something about the macOS setup, not this change.